test/fs.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const metatests = require('metatests');
+const metasync = require('metasync');
+const common = require('..');
+const fs = require('fs');
+const path = require('path');
+const stream = require('stream');
+
+const testData =
+  'JySAF1figbONWxAeeoGhDXeI7df25GKXKPcyCl63ZjIv5HZGQH8yfFeHGGaSJ9zDMP0CwtgX2madIQkFmuq38SNhcKcGQ';
+const absoluteLink = __dirname.replace('test', 'lib');
+const srcFiles = [
+  'srcDir',
+  'srcDir/dir1',
+  'srcDir/dir2',
+  'srcDir/file1',
+  'srcDir/link',
+  'srcDir/dir1/subdir1',
+];
+
+const dstFiles = [
+  'dstDir',
+  'dstDir/dir1',
+  'dstDir/dir2',
+  'dstDir/file1',
+  'dstDir/link',
+  'dstDir/dir1/subdir1',
+];
+
+class ToUpperCase extends stream.Transform {
+  constructor(options = {}) {
+    options.decodeStrings = false;
+    super(options);
+  }
+  _transform(chunk, encoding, callback) {
+    if (encoding !== 'utf8') {
```

Comment:
According to [stream documentation](https://nodejs.org/api/stream.html#stream_transform_transform_chunk_encoding_callback), it can be done much simpler.
```js
if (encoding === 'utf8') callback(null, chunk.toUpperCase());
else callback(new Error('Only utf8 is supported'));
```

Reply:
To be honest I just took Alexey Orlando example stream.
Ok I will remake it.

Reply:
You cannot just copy the code of others without permission or giving them credits for that. In case this is necessary you should check if [the licence](https://choosealicense.com/appendix/) allows that and give credit to the original author. If this is not done it can create problems with copyright. Furthermore, this is not a complex code and it shouldn't be copied.